index.js

```diff
@@ -0,0 +1,115 @@
+const URL = 'https://trektravel.herokuapp.com/trips';
+
+
+const reportStatus = (message, alertClass) => {
+    $('#status-message').removeClass();
+    $('#status-message').addClass(alertClass);
+    $('#status-message').html(message);
+};
+
+const reportError = (message, errors) => {
+    let content = `<p>${message}</p>`
+    content += "<ul>";
+    for (const field in errors) {
+      for (const problem of errors[field]) {
+        content += `<li>${field}: ${problem}</li>`;
+      }
+    }
+    content += "</ul>";
+    reportStatus(content, "alert alert-warning");
+  };
+
+const clearForm = () => {
+    $('#trip-form').get(0).reset();
+};
+
+
+
+const loadTrips = () => {
+    reportStatus('Loading trips...');
+
+    const tripList = $('#trips-list');
+    tripList.empty();
+
+    axios.get(URL)
+        .then((response) => {
+            reportStatus(`Successfully loaded ${response.data.length} trips`, "alert alert-success");
+            response.data.forEach((trip) => {
+                tripList.append(`<li id='${trip.id}'><a href=#>${trip.name}</a></li><hr />`);
+                $(`#${trip.id}`).click(function () {
+                    showTrip(trip.id);
+                });
+            });
+        })
+        .catch((error) => {
+            reportStatus(`Encountered an error while loading trips: ${error.message}`);
+            console.log(error);
+        });
+};
+
+const showTrip = (index) => {
+    $('#status-message').removeClass();
+    $('#status-message').html('');
+    
+    axios.get(URL + `/${index}`)
+        .then((response) => {
+            $('#form-section').removeClass('hidden');
+            $('#trip-form').submit(function (event) {
+                $('#tripID').val(response.data.id);
```

Comment:
The work being done here is complex enough that it might be worthwhile to break it out into its own function.